lib/transaction.js

```diff
@@ -18,14 +20,18 @@ function mixinTransaction(PostgreSQL) {
    * @param cb
    */
   PostgreSQL.prototype.beginTransaction = function(isolationLevel, cb) {
+    var connector = this;
     debug('Begin a transaction with isolation level: %s', isolationLevel);
     this.pg.connect(function(err, connection, done) {
       if (err) return cb(err);
       connection.autorelease = done;
       connection.query('BEGIN TRANSACTION ISOLATION LEVEL ' + isolationLevel,
         function(err) {
           if (err) return cb(err);
-          cb(null, connection);
+          var tx = new Transaction(connector, connection);
+          tx.txId = uuid.v1();
+          connection.txId = tx.txId;
+          cb(null, tx);
```

Comment:
@zbarbuto after I fixed `if(err) return done(err)`, now we get the real error:
```shell
 1) transactions bulk should work with bulk transactions:
     Error: read ECONNRESET
      at exports._errnoException (util.js:907:11)
      at TCP.onread (net.js:558:26)
```
I suspect it's because here we callback the `tx`, then when we do `tx.commit` to release it, but we don't close/release the `connection` in L25...
The old code callback `connection` so it doesn't have this error.